### ISSUE TITLE
Recycler Unit CBM. Possibly not working correctly. Suggestion for change

### ISSUE THREAD

**Issue by IdleSol** [CleverRaven/Cataclysm-DDA#74110]
### Is your feature request related to a problem? Please describe.

Let me say right off the bat, I don't know exactly how it works. My observations, have shown the following:
1. A character without the module, standing in one place, spends 2700 kcal in 24 hours.
2. A character with the module, standing in one place, spending 1800 kcal in 24 hours.

*Note. I created a character.  I set his caloric reserve to 120,000 kcalories. I reset his stomach and gut reserve (=0). And waited 4 times for 6 hours each. After 12 hours, reset his thirst and hunger. On a separate note, the character had traits, so this is not a pure experiment. Among the traits were thirst and metabolism. But I was interested in relative values, not exact values.*

I assume that this module reduces the calorie expenditure of the character's actions. And somehow interacts with water (I haven't checked).

In fact, it's an energy-saving module. It reduces costs by 33%. Probably more or less, as I only tested a stationary character.

>A series of filters and processors that is implanted in the user's digestive system, allowing them to reclaim waste liquid and, to a lesser degree, nutrients. The net effect is a greatly reduced need to eat and drink.

The description tells us that the module allows us to get more water and energy from digestive waste. There's nothing about reducing costs here. And it's not supposed to.

Because the character hasn't eaten anything. There's nothing for the module to work with. But 900 kcal came from somewhere. Energy from the air.

*Note. Here is the finished module for the Exodii labyrinth.*

What's the threshold for death by starvation? I'm guessing 20k. That's 100k kcal difference, converted into an extra 33k kcal from the air.

### Solution you would like.

I would like this module to work with and depend on incoming water and food specifically.

The simplest implementation, increasing the water and calorie values for the food eaten. But there is already such a module: Expanded Digestive System CBM. But it works only with calories.

And it's not a good fit for a module that has to deal with refuse.

So I want to suggest a slightly different option. We have a reservoir of stored energy in the gut. The energy from that goes into the total reservoir. Every 30 minutes, 37.5 kilocalories are transferred from the gut to the total reservoir. (To be more precise 37 alternates with 38, I assume 37.5 is used and rounded).

Considering the 33% efficiency. Why wouldn't every third kilocalorie go back to the gut (or stomach if it's responsible for feeling hungry).

Then of the 37.5 kcal, 37.5 go into the total reservoir, and 37.5 / 3 = 12.5 kilocalories are added to the gut reservoir.

Perhaps through a random number between 1 and 3. Instead of decreasing the value use an increase? Or just increases the whole batch by 1/3? Then the total reservoir receives not 37.5, but 37.5 + 12.5 = 50 kcal. This only wastes 37.5 kcal.

Water is more difficult, we don't have a separate reservoir for stored water. We just have thirst, which can be greater or less than 0. Where a value less than 0 is the value of stored water (?). Similarly, if the value is less than 0, then there is a 1 in 3 chance of increasing (actually decreasing, since the value is negative) the water parameter.

So we will link the work of this module to the food we consume. It will still provide energy and water from the air, but it will not do so indefinitely.

### Describe alternatives you have considered.

The standard alternative, change nothing.

**Comment by GuardianDll** [comment]
Is there any meaningful difference between how it works and how you suggest it works?

**Comment by IdleSol** [comment]
Yes. If you don't have anything in your stomach/guts, nothing recovers.

**Comment by PatrikLundell** [comment]
If you were to install the CBM into a morbidly obese character who went or a weight reduction regimen with minimal calorie intake to just provide vitamins the current implementation would reduce the weight loss rate significantly, while the suggested change would keep the rate almost intact (and fully intact if vitamin pills were used exclusively).

It's an unlikely case in practice, though, unless you were role playing a character with an eating disorder.

**Comment by GuardianDll** [comment]
it is not from the air, it is just assumed it's from the food you consumed previously
if you want, you can reflavor it for something like "makes you metabolism better duh", because returning processed food back to the guts with 1/3 chance would be absolutely no different from just increasing amount of calories for 33%
Which, as you say, Expanded Digestive System already does

**Comment by Kamejeir** [comment]
I'd love for the CBM to make my character not be hungry as often.
